boards/FSM/Src/App/App_Brake.c

```diff
@@ -1,77 +1,141 @@
 
     return brake;
 }
-
 void App_Brake_Destroy(struct Brake *brake)
 {
-    App_InRangeCheck_Destroy(brake->pressure_in_range_check);
+    App_InRangeCheck_Destroy(brake->front_pressure_in_range_check);
+    App_InRangeCheck_Destroy(brake->rear_pressure_in_range_check);
     free(brake);
 }
 
-struct InRangeCheck *App_Brake_GetPressureInRangeCheck(const struct Brake *const brake)
-{
-    return brake->pressure_in_range_check;
-}
-
+// act
 bool App_Brake_IsBrakeActuated(const struct Brake *brake)
 {
     return brake->is_brake_actuated();
 }
 
-bool App_Brake_IsPressureSensorOpenOrShortCircuit(const struct Brake *brake)
+// pressures
+float App_Brake_GetFrontPSI(const struct Brake *brake)
+{
+    return brake->get_front_pressure_psi();
+}
+float App_Brake_GetRearPSI(const struct Brake *brake)
 {
-    return brake->is_pressure_sensor_open_or_short_circuit();
+    return brake->get_rear_pressure_psi();
+}
+bool App_Brake_PressureElectricalFault(const struct Brake *brake)
+{
+    return brake->front_pressure_sensor_ocsc() || brake->rear_pressure_sensor_ocsc();
+}
+struct InRangeCheck *App_Brake_GetPressureInRangeCheck(const struct Brake *const brake)
+{
+    return brake->front_pressure_in_range_check;
 }
 
-bool App_Brake_IsPressureSensorOpenCircuit(const struct Brake *brake)
+// pedal
+float App_Brake_GetPedalTravel(const struct Brake *brake)
+{
+    return brake->get_pedal_travel();
+}
+bool App_Brake_PedalSensorAlarm(const struct Brake *brake)
 {
-    return brake->get_pressure_psi() < BRAKE_PRESSURE_OC_THRESHOLD;
+    return brake->pedal_travel_sensor_ocsc();
 }
 
-bool App_Brake_IsPressureSensorShortCircuited(const struct Brake *brake)
+// broadcast
+void App_Brake_Broadcast(const struct FsmWorld *world)
 {
-    return brake->get_pressure_psi() > BRAKE_PRESSURE_SC_THRESHOLD;
+    struct FsmCanTxInterface *can_tx = App_FsmWorld_GetCanTx(world);
+    struct Brake *            brake  = App_FsmWorld_GetBrake(world);
+
+    // Brake Pedal Value
+    App_CanTx_SetPeriodicSignal_BRAKE_PEDAL_PERCENTAGE(can_tx, brake->get_pedal_travel());
+
+    // Brake Actuation Value
+    uint8_t CANMSGS_FSM_BRAKE_BRAKE_IS_ACTUATED = brake->is_brake_actuated()
+                                                      ? CANMSGS_FSM_BRAKE_FLAGS_BRAKE_IS_ACTUATED_TRUE_CHOICE
+                                                      : CANMSGS_FSM_BRAKE_FLAGS_BRAKE_IS_ACTUATED_FALSE_CHOICE;
+    App_CanTx_SetPeriodicSignal_BRAKE_IS_ACTUATED(can_tx, CANMSGS_FSM_BRAKE_BRAKE_IS_ACTUATED);
+
+    // Pressure Values, with range check
+    App_SetPeriodicCanSignals_InRangeCheck(
+        can_tx, brake->front_pressure_in_range_check, App_CanTx_SetPeriodicSignal_FRONT_BRAKE_PRESSURE,
+        App_CanTx_SetPeriodicSignal_BRAKE_FRONT_PRESSURE_OUT_OF_RANGE,
+        CANMSGS_FSM_NON_CRITICAL_ERRORS_BRAKE_FRONT_PRESSURE_OUT_OF_RANGE_OK_CHOICE,
+        CANMSGS_FSM_NON_CRITICAL_ERRORS_BRAKE_FRONT_PRESSURE_OUT_OF_RANGE_UNDERFLOW_CHOICE,
```

Comment:
These are not the correct choices for this sensor. Should be in-range or out-of-range. There's no flow in the brakes

Reply:
fixed 7603832